playbooks/common/openshift-docker/config.yml

```diff
@@ -0,0 +1,85 @@
+        local_facts:
+          deployment_type: "{{ openshift_deployment_type }}"
+
+- name: Configure docker for additional repositories
+  hosts:
+    - masters
+    - nodes
+  handlers:
+    - name: restart docker
+      service: name=docker state=restarted
+  tasks:
+    # TODO -- currently this is sane to do on masters since we require
+    # that a master be part of the sdn which requires node and docker packages
+    # When we fix the SDN to work without a running node we should clean this up
+    - yum: name=docker state=installed
+    - stat: path=/etc/sysconfig/docker
+      register: docker_check
+    - set_fact:
+        docker_additional_registries: "{{ lookup('oo_option', 'docker_additional_registries')
+                                        | oo_split() | union(['registry.access.redhat.com'])
+                                        |  difference(['']) }}"
+      when: openshift.common.deployment_type in ['enterprise','atomic-enterprise','openshift-enterprise']
+    - set_fact:
+        docker_additional_registries: "{{ lookup('oo_option', 'docker_additional_registries')
+                                      | oo_split() | difference(['']) }}"
+      when: openshift.common.deployment_type not in ['enterprise','atomic-enterprise','openshift-enterprise']
+    - name: Add personal registries
+      lineinfile:
+        dest: /etc/sysconfig/docker
+        regexp: '^ADD_REGISTRY=.*$'
+        line: "ADD_REGISTRY='{{ docker_additional_registries
+                            | oo_prepend_strings_in_list('--add-registry ') | join(' ') }}'"
+      when: docker_check.stat.isreg and docker_additional_registries
+      notify:
+        - restart docker
+
+    - name: Block registries
+      lineinfile:
+        dest: /etc/sysconfig/docker
+        regexp: '^BLOCK_REGISTRY=.*$'
+        line: "BLOCK_REGISTRY='{{ lookup('oo_option', 'docker_blocked_registries') | oo_split()
+                              | oo_prepend_strings_in_list('--block-registry ') | join(' ') }}'"
+      when: docker_check.stat.isreg and
+          lookup('oo_option', 'docker_blocked_registries') != ''
+      notify:
+        - restart docker
+
+    - name: Grant access to additional insecure registries
+      lineinfile:
+        dest: /etc/sysconfig/docker
+        regexp: '^INSECURE_REGISTRY=.*'
+        line: "INSECURE_REGISTRY='{{ lookup('oo_option', 'docker_insecure_registries') | oo_split()
+                              | oo_prepend_strings_in_list('--insecure-registry ') | join(' ') }}'"
+      when: docker_check.stat.isreg and
+          lookup('oo_option', 'docker_insecure_registries') != ''
+      notify:
+        - restart docker
+
+# If we don't pre-fetch images containerized installations fail more often than not
+- name: Pre-fetch images for containerized installs
+  hosts:
+    - masters
+    - nodes
+  tasks:
+    - name: Pull base image
+      command: docker pull {{ openshift.docker.master_image }}
+      when: is_containerized | bool
```

Comment:
Considering the uncertainty of the PR, I would almost rather we built a module to just handle the unauthenticated docker pull use case that we need.